connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -601,3 +602,10 @@ message Unpivot {
   // (Required) Name of the value column.
   string value_column_name = 5;
 }
+
+// Randomly splits this Dataset with the provided weights.
+// Note: this message is just a wrapper for input relation.
+message RandomSplit {
+  // (Required) The input relation.
+  Relation input = 1;
```

Comment:
hmmm I am thinking you do not need this message. Maybe you only need add a `bool` to the Sample message to indicate this is a case of random split?

cc @zhengruifeng @cloud-fan

Reply:
This method is also possible, but it seems to be somewhat coupled.

Reply:
I might be missing some context, please help me :)

It looks like you want to express that RandomSplit can never occur outside of Sample. So the wrapping would be something like

```
Sample(RandomSplit(input))
```

In the current approach for the Relations all relations are more or less valid top level objects that can be used independently. I feel that in this case RandomSplit should rather be an element of the Sample message.